packages/myst-cli/src/process/mdast.ts

```diff
@@ -334,6 +338,41 @@ export async function postProcessMdast(
   if (checkLinks) await checkLinksTransform(session, file, mdast);
 }
 
+/**
+ * Write widget state to file in public folder
+ */
+export function transformWidgetsToFile(
```

Comment:
Steve and I chatted about this - there is some duplication between this and the notebook `outputs` caching. These also follow a slightly different pattern than other on-disk caching that is saved to the `/cache` folder. Definitely potential to normalize and deduplicate code... but for now, I think this standalone function is fine.